src/parser/mod.rs

```diff
@@ -484,8 +488,18 @@ impl<'a> Parser<'a> {
             }
 
             let statement = self.parse_statement()?;
+            expecting_statement_delimiter = match &statement {
+                Statement::If(s) => match &s.if_block.conditional_statements {
+                    // the `END` keyword doesn't need to be followed by a statement delimiter, so it shouldn't be expected here
+                    ConditionalStatements::BeginEnd { .. } => false,
+                    // parsing the statement sequence consumes the statement delimiter, so it shouldn't be expected here
+                    ConditionalStatements::Sequence { .. } => false,
+                },
+                // Treat batch delimiter as an end of statement, so no additional statement delimiter expected here
```

Comment:
will this be replaced by similar solution as in #1808?

Reply:
I suspect so... need to investigate. (I need to rebuild my mental model on this first 😆).